index.js

```diff
@@ -19,26 +21,44 @@ util.inherits(CpFileError, NestedError);
 
 CpFileError.prototype.name = 'CpFileError';
 
+function statSource(src, progress) {
+	if (!progress.enabled) {
+		return Promise.resolve();
+	}
+	return fsP.stat(src).then(function (stat) {
+		progress.stat = stat;
+	}).catch(function (err) {
+		Promise.reject(new CpFileError('NO_ENTRY', err));
+	});
+}
+
 module.exports = function (src, dest, opts) {
 	if (!src || !dest) {
 		return Promise.reject(new CpFileError('`src` and `dest` required'));
 	}
 
-	opts = objectAssign({overwrite: true}, opts);
+	opts = objectAssign({overwrite: true, onProgress: function () {}}, opts);
 
-	return new Promise(function startRead(resolve, reject) {
-		var read = fs.createReadStream(src);
+	var progressEnabled = typeof opts.onProgress === 'function';
```

Comment:
This is also faulty as you assign a default value in the objectAssign call, so this will always be `true`.